server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -433,7 +433,11 @@ public synchronized <T> void initializeAndWatch(Setting<T> setting, Consumer<T>
         assert setting.getProperties().contains(Setting.Property.Dynamic)
             || setting.getProperties().contains(Setting.Property.OperatorDynamic) : "Can only watch dynamic settings";
         assert setting.getProperties().contains(Setting.Property.NodeScope) : "Can only watch node settings";
-        consumer.accept(setting.get(settings));
+
+        // this mimics the combined settings of last applied and node settings, without building a new settings object
```

Comment:
With this comment I understand you are trying to avoid doing 
`Settings.builder().put(this.settings).put(this.lastSettingsApplied).build()` like in other functions, correct?

Is there an advantage? Is it correct? (is `lastSettingsApplied` always a superset of `settings`?)

Reply:
> Building a combined Settings object is heavyweight. It requires combing the internal maps that each Settings object contains.

I understand that, but I was wondering if it really matters in this case - I might be mistaken, but this is an initialization method, so probably it's not on a hot path.

As long as it's correct I like the optimization though.